src/linalg.jl

```diff
@@ -62,6 +62,11 @@ end
 #--------------------------------------------------
 # Matrix algebra
 
+@inline _adjointtype(f, ::Type{T}) where {T} = Base.promote_op(f, T)
```

Comment:
This looks fine but could you add a comment here explaining the idea behind `_adjointtype`?